test/envtest/konnect_entities_key_test.go

```diff
@@ -71,50 +73,145 @@ func TestKongKey(t *testing.T) {
 	t.Log("Setting up a watch for KongKey events")
 	w := setupWatch[configurationv1alpha1.KongKeyList](t, ctx, cl, client.InNamespace(ns.Name))
 
-	t.Log("Creating KongKey")
-	createdKey := deploy.KongKeyAttachedToCP(t, ctx, clientNamespaced, keyKid, keyName, cp)
+	t.Run("without KongKeySet", func(t *testing.T) {
+		t.Log("Creating KongKey")
+		createdKey := deploy.KongKeyAttachedToCP(t, ctx, clientNamespaced, keyKid, keyName, cp)
+
+		t.Log("Waiting for KongKey to be programmed")
+		watchFor(t, ctx, w, watch.Modified, func(c *configurationv1alpha1.KongKey) bool {
+			if c.GetName() != createdKey.GetName() {
+				return false
+			}
+			return lo.ContainsBy(c.Status.Conditions, func(condition metav1.Condition) bool {
+				return condition.Type == conditions.KonnectEntityProgrammedConditionType &&
+					condition.Status == metav1.ConditionTrue
+			})
+		}, "KongKey's Programmed condition should be true eventually")
+
+		t.Log("Waiting for KongKey to be created in the SDK")
```

Comment:
Nit: the log here is not 100% accurate because when the object already got `Programmed` (checked above) then it means that the SDK operation has already taken place.

Perhaps something like this would be more accurate: 

```suggestion
		t.Log("Checking SDK KongKey operations")
```